### PR TITLE
Update pythonpackage.yml

### PR DESCRIPTION
**Pull request by koaning**
- i am turning on github actions
- i am turning off travis

### CODE REVIEW

**Comment by koaning** [comment]
- removed pathos
- added gitpod

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Dependencies
+      run: |
+        pip install -e ".[dev]"
+        python setup.py develop
```
Comment (rogiervandergeer) [inline .github/workflows/pythonpackage.yml]: Isn't `python setup.py develop` a duplicate of `pip install -e`?

File: .github/workflows/pythonpackage.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Dependencies
+      run: |
+        pip install -e ".[dev]"
```
Comment (rogiervandergeer) [inline .github/workflows/pythonpackage.yml]: I think we should also test `evol` without installing the `dev` part.

File: .github/workflows/pythonpackage.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.6, 3.7]
```
Comment (rogiervandergeer) [inline .github/workflows/pythonpackage.yml]: Is 3.8 already available here? Perhaps `3.8-dev`?

File: .github/workflows/pythonpackage.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Dependencies
+      run: |
+        pip install -e ".[dev]"
+        python setup.py develop
+    - name: flake8
+      run: |
+        pip install flake8
```
Comment (rogiervandergeer) [inline .github/workflows/pythonpackage.yml]: Better run `python setup.py flake8`. No need to install flake8 then.

File: .github/workflows/pythonpackage.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        flake8 
+    - name: Test with pytest
+      run: |
+        pip install pytest
```
Comment (rogiervandergeer) [inline .github/workflows/pythonpackage.yml]: Better run `python setup.py test`. No need to install pytest or any of the other test dependencies.

File: .gitpod.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+tasks:
+  - init: pyenv local 3.7.2 && pip install -e ".[dev]"
+    command: python setup.py develop
```
Comment (rogiervandergeer) [inline .gitpod.yml]: No need to run `python setup.py develop`, as it is already installed by the previous line.

File: .github/workflows/pythonpackage.yml (hunk 3 above)
Comment (koaning) [reply .github/workflows/pythonpackage.yml]: Alas, no. 

```
Available versions:

2.7.16 (x64)
3.5.7 (x64)
3.6.9 (x64)
3.7.4 (x64)
##[error]Node run failed with exit code 1
```

Is this a reason to keep travis around? I think 3.8 is going to take a while before people will use it. I am fine with ignoring 3.8 for now, you?

**Comment by koaning** [comment]
I think I am missing something. How can you run flake8/pytest without installing it? I also think that we *must* install multiprocess for pytest to run. I'd like to install sphinx as well since doc rendering issues might be a thing.

File: .github/workflows/pythonpackage.yml (hunk 3 above)
Comment (rogiervandergeer) [reply .github/workflows/pythonpackage.yml]: Travis doesn't have it either. It was worth trying, but no problem if we don't have it.

File: .github/workflows/pythonpackage.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: install dev deps
+      run: pip install -e ".[dev]"
+    - name: develop
+      run: python setup.py develop
```
Comment (rogiervandergeer) [inline .github/workflows/pythonpackage.yml]: This isn't needed:
`evol 0.4.1 is already the active version in easy-install.pth`

**Comment by rogiervandergeer** [comment]
> I think I am missing something. How can you run flake8/pytest without installing it? I also think that we _must_ install multiprocess for pytest to run. I'd like to install sphinx as well since doc rendering issues might be a thing.

The `setup.py` handles the installation of the `setup_requires` and `tests_require` dependencies when you run `python setup.py <test/flake8>`. See https://blog.godatadriven.com/setup-py. You may need to shuffle the requirements in the `setup.py`. In any case, if `multiprocess` is absolutely required to use `evol`, we need to add it to the `install_requires`.

File: .github/workflows/pythonpackage.yml (hunk 7 above)
Comment (koaning) [reply .github/workflows/pythonpackage.yml]: Removed it.

**Comment by koaning** [comment]
I've shuffled things around. This should be a lot better. Happy to report that the CI now takes less than a minute.

**Comment by rogiervandergeer** [comment]
You're missing some newlines at the end of files, but other than that this looks good!

**Pull request merged**